Client/Networking/NetworkSession.swift

```diff
@@ -62,6 +62,7 @@ extension URLSession: NetworkSession {
     return self.dataTaskPublisher(for: urlRequest)
       .map({ ($0, $1) })
       .mapError({ $0 as Error })
+      .receive(on: DispatchQueue.main)
```

Comment:
@Brandon-T I assume the prior closure-based version used to jump to main for the completion handler since this crash is a regression, so I think receiving on `main` for all network session tasks is probably fine. Let me know if there's any case where this doesn't work